### DISCUSSION TITLE
373 Latch on Address lines A8-A15

### DISCUSSION THREAD

**Discussion by TheEccentricGenius** [TheEccentricGenius/HD6303RP_EVB#5] [General]
@KenWillmott  question
> I see you put a 373 latch U7 on address lines A8-A15. Just wondering what you have in mind, because those signals are valid early in the cycle, and remain valid until the falling edge of E.

**Comment by TheEccentricGenius** [comment]
@KenWillmott I have moved the discussion from the Announcements  category.

**Comment by TheEccentricGenius** [comment]
I thought it was necessary by this schematic, [373 latch on address lines A8-15.](https://github.com/KenWillmott/HD6303EVB/discussions/8#discussioncomment-6385674)

A quick glance at the Hitachi 6301/3 handbook, shows it would only be necessary for noise immunity IIRC.

It can be removed if you please, I won't care it will be one less chip on the board plus to go wrong.

**Comment by KenWillmott** [comment]
Oh, I see. That extra latch is only needed for the larger than 32 RAM IC's. In your case you don't need it. It does provide a buffer, but if you want to use a buffer there, a 74HC541 is a better fit. As long as it's basically an SBC, the buffering is not strictly necessary.

The schematic you reference got a little obfuscated because there are so many address lines. Just for clarification, are you planning to use 32k or 512k NVRAM?

**Reply by TheEccentricGenius** [reply]
So, sorry I didn't notice the 512k NVRAM in the schematic till now.
I'll remove the extra latch and update the schematic tomorrow hopefully.

I am planning on going with the 32K RAM currently.....512K is nice, but I am not sure about using currently.

**Comment by KenWillmott** [comment]
The banked 32k RAM is the simplest way with this circuit. You could even simplify even further by omitting the second RAM entirely, and connecting the NVRAM controller to the main SRAM at $0000-$7FFF. A lot of designs like the Searle/Tranter board don't have any more than that. The 512k RAM expansion has become my main focus this week, it's really hard to do without a proliferation of glue logic. If it becomes too complex, I will consider putting it on an expansion board.

**Reply by TheEccentricGenius** [reply]
Yes, the 512k RAM glue logic looks a little more complex....I may do one SRAM, however I like the idea of having two banks.

Sorry, didn't get around to fixing the latch circuitry.

**Comment by TheEccentricGenius** [comment]
I have completed the change, it looks a bit messy in the schematic though.

The NVBANK issue was also addressed....I ended up using one of the 8 IO pins I had set apart.

**Comment by KenWillmott** [comment]
It would be helpful to export the current schematic as a PDF and upload, for quick looks...

**Reply by TheEccentricGenius** [reply]
Done, except the dodgy export KiCAD appears to of done.

**Comment by KenWillmott** [comment]
Thanks, yes it's strange what it did to the gates. One important change is needed, to be M8 bus compliant, the select BSEL* should not be gated with E as it was in the first iteration. That should happen on any peripheral card, not on the main board. For any on board peripherals that need E gating, you can remove U9A from the input A1 of U10, instead connect it directly to A12. Then move the outputs from U10 00,01 to outputs 02,03. If you need E gating for on board peripherals you have two spare NAND gates available to do that, U9A and U1A. You would add that circuit to one of the U10 outputs like this:
![image](https://github.com/TheEccentricGenius/HD6303RP_EVB/assets/17345651/846467be-9712-4360-960a-a32b9f85b2d3)

The Searle/Tranter design doesn't require E gating glue logic because the ACIA has one positive level enable input that is connected directly to E. Some peripheral chips have such an enable, some don't.

**Reply by TheEccentricGenius** [reply]
This [decode](https://github.com/KenWillmott/HD6303EVB/discussions/9#discussioncomment-6920996) will work with the HD6303RP board?

**Comment by KenWillmott** [comment]
It can work with the HD6303RP. That circuit is from a new HD6303Y board, I call the "Maximus". The main difference is the memory complement. It uses a 32kb EEPROM instead of the 8kb on the EVB, and has one SRAM instead of two. The one SRAM can be 32kb, 512kb, or even 1Mb. If you leave out the memory expansion latch, it can still work with the 32kb SRAM. Versus the EVB decode, it uses one more IC, the 74HC20. If board space is at a premium, that could factor into a decision about which decode to use. But, it is an improvement over the original circuit.

I've been working on it full time, but got sidetracked when the game board PCBs arrived. I meant to push it out to a repository of its own, but I wanted to polish some of the rough edges first. My time is now dedicated to testing the game board, so that I can iron out any problems, and order more parts so I can build more.

With the small modification that I mentioned, either circuit will correctly drive the M8 bus expansion connector. So, the answer depends on what memory you want to use, and which memory map looks more useful to you.

I updated the new memory map to show the two EEPROM areas separately. Basically the upper 4k becomes the boot, system and monitor ROM:
Memory Map:

0000-7FFF NVRAM 32k
8000-DFFF EEPROM 24k
E000-E3FF main bus select 0 1K
E400-E7FF main bus select 1 1K
E800-EBFF on board select 0 1K
EC00-EFFF memory latch select 1K
F000-FFFF EEPROM 4K

Additionally, if more on board peripherals exist besides the memory latch, it's evident from the game board decoding design that a single 74HC138 will do the job by itself, of selecting those peripherals in, for example, the on board select. OBSEL* can be used bare, to select only one peripheral, or it can be multiplexed out with an HC138.

If you have only one expansion bus connector and also no extended RAM so no latch,  then there are still three more selects you can use for on board peripherals without any additional HC138. The only catch is that you still need some logic to gate the signal with E. That is what the HC138 circuit does on the game board, but you can just as well implement it with any logic such as NAND gates if it's more expedient.

The original EVB design favours the on board decoding, not the bus. after removing E gating from the select signals, the EVB now requires external logic such as on the game board, to access peripherals. However at least the new design manages to scrape up just enough gates to provide one on board select gated with E. As I mentioned before, you can squeak around the gating problem with some peripheral chips, since they have inputs that you can connect directly to E.

The new design has three non-gated, and one gated, memory selects. The original one, two gated selects. When modified, two non-gated selects.

Either an EVB with corrected select logic, or a Maximus, is M8 bus compliant so will drive the game board and other boards correctly. The only difference in that regard, is that the EVB loses the logic to drive a latch with the bus signal correction. So only the new design can support more than 32k NVRAM. The EVB can not unless some logic is added to it.

**Comment by KenWillmott** [comment]
I'm posting the new system schematic on my repo...

**Reply by TheEccentricGenius** [reply]
Thanks.

**Comment by TheEccentricGenius** [comment]
Is this how you were trying to describe how to make the bus M8 compliant?
![image](https://github.com/TheEccentricGenius/HD6303RP_EVB/assets/137060094/93b82e77-89e1-4352-93ab-f6a95c9891d9)

**Comment by KenWillmott** [comment]
Basically, yes. Removing E gating from U10A actually opens up four possible selects on the outputs Q0-Q3. For any device that requires E gating, it has to be added after this decoding, I do it with a NAND and NAND inverter as below. If there are no on board peripherals, or the on board peripherals have their own E gating, you can leave out the two NAND circuit U5A-U8A.
![image](https://github.com/TheEccentricGenius/HD6303RP_EVB/assets/17345651/8db908f7-e8a1-4cc7-8e2a-5a201016c5e0)

The E gated signal GSEL* is required for the on board address latch that is used to address a 512kb NVRAM. If you have only a 32kb NVRAM, it's not needed. In that case, you can arbitrarily assign Q0-Q3 to individual bus connector selects, or to any on board peripherals that you have (as long as they have their own independent E gating).

Just as an observation, if I recall correctly, your memory select circuit used only 3 NAND gates, so one is left over, with two 74HC00 packages, you have enough gates to implement GSEL* in the decode block.

**Reply by TheEccentricGenius** [reply]
So, would it matter if I just put GSEL* on for a future board, but it doesn't do anything on this board?

Or should I just move to 512kb NVRAM?
What other complexities will need to be done for NVRAM?

**Comment by KenWillmott** [comment]
One possibility for you, the pinouts of a 32kb and 512kb NVRAM are almost the same. So you could provide a socket that could accept either one, using a jumper header to configure the few signals that are different. I need to check how many but I think it's only one or two.

Reflecting on this, I may fold the same idea into the 6303Y evolution.

You need at least 6 NAND gates no matter what, so the extra 2 gates that you get with two IC's are "free" to provide a GSEL* signal (and bring out to a header for experimentation). You could equip the board with a socket for the latch, but leave it unpopulated and install a 32k RAM that in the NVRAM socket. Or else, plug a latch into the empty socket, change the jumpers, and place a 512kb RAM there. All the other circuits besides the latch and a few jumpers are the same in either case.

I think you will later regret not accommodating 512kb. The reason, there is actually only a small difference in complexity between the 32k and 512k circuits. The cost difference, or differing needs for different applications, is solved by making the board compatible with either part. However, the simplest circuit with only 32kB NVRAM, in two 16kb banks may use an I/O port to select the bank as you mentioned before. If you want to just go with 32kb, you can still provide GSEL* on a pin, to use for breadboard prototypes. It saves having to re-create the hardware on the proto board. Another way is an on board jumper to select between E gated and not gated BSEL* signals. But I don't want it to gravitate into end use boards using gated signals, because it isn't compatible with optimal timing on the bus, between multiple cards. It is always useful in playing with new peripheral chips.

The difference between using GSEL* on and off the main board is, that it can't be gated with additional logic, without delaying the E clock as it is seen by the peripheral device. In the case of the GSEL* logic, E is applied to the very last gate in the signal chain and so has minimal delay. Thus if there is no on board device, GSEL* on the bus can select one peripheral device on the bus, but that runs into timing delay problems if multiple devices should be selected on each board. For example, the game board has four.

Assuming that a system has some E gated select signal available for it, a latch IC can be the only additional component in a simplified address system. Data accessed in the 16k paged area, is separate from both ROM and the base 32k RAM, and software can page reliably as long as the code that's running doesn't reside within it. In contrast, the newer Maximus design sacrifices that convenience in order to use only one RAM chip. In the new design, paged access have to be mediated by a byte level handler routine, which is designed specifically not to use any RAM in its short run of codes. It only transfers between registers and locations in extended (512kb or 1Mb) NVRAM. So it can be called by a file system handler to transparently access bytes in extended RAM. It might gain some versatility from code using the 6303's internal RAM, but that is a long story. But with both types of decode, the only additional component is a latch.

Another dual purpose thing you could do, is assign latch bits to digital output pins. Because, the few upper bits of the latch aren't used by the extended memory latch anyway. At least, there could be an on board status LED, like the "blinky" on the Arduino. Using the 32kb memory only, you could use all 8 bits for output. If you bring the lines to headers, and provide configuration options, there could be some mix of output, and some or none extended memory addressing bits.

**Reply by TheEccentricGenius** [reply]
I'll think this over and re-read your post to be sure I've seen everything.

**Reply by TheEccentricGenius** [reply]
I only need a jumper to select between A17 and VCC, otherwise the pinout is compatible.
I can feel I'll regret it so I'll add at least 512K memory capability, will I need the 4 input NAND gate though?

I'll definitely try to assign upper latch bits to digital pins.

**Comment by KenWillmott** [comment]
The NAND gates come in packages of 4. Three are used in the processor schematic, to produce memory read and write signals. 3 more are used in the basic decode circuit. So you have a minimum of 6. That means, you have 2 left over. You could actually use them for anything, using them to derive a GSEL* is only a suggestion. But it dovetails nicely with the 512k RAM concept, since you need one full on board decode for that, to drive the latch.

By the way, I just noticed that there is an intermediate size NVRAM that also fits with the jumper mod scheme, a 128k part. The ones I saw were also incredibly cheap. So I'm going to check the pinout on that one too. But it's probably just got the same pinout as the 512k, just some address pins not used.

There's a big sea change on my end, but won't affect your project, the processor and main board are going to be split, so different processors can be plugged into the board. :) You will see it come out in dribs and drabs next week. Actually, it won't change my circuits much either, just the PCB configuration.

**Comment by KenWillmott** [comment]
The 128kb RAM is an AS6C1008-55. I found, with that chip pin 30 is CS2, which can be tied to Vcc. In that case, either a 32kb or 128kb NVRAM would drop right into the socket, no jumpers needed. It's only with the AS6C4008, the 512kb part, that pin 30 needs a jumper header.

**Reply by TheEccentricGenius** [reply]
I'll post a schematic here for review when I've done adding and modifying the current circuit.

**Comment by KenWillmott** [comment]
Good. I found, I can't build my system board without resorting to SMD, so the memory footprint issue is different on that one at least. It's coming along as mentioned, I think I'll be able to get something in fab by the end of this week. I found out, it is not terribly expensive to run multiple board layouts on the same panel. So the next order will probably have two, "MAD" board which is memory and decode with 4 bus slots, and one 6809 processor board. A 6502 processor board will follow later. My existing 6303Y board can be easily modified to act as a processor only board, so I don't need to design any of those just now. Thanks for your patient efforts.

**Comment by KenWillmott** [comment]
The latest board is beta release:
https://github.com/KenWillmott/SST-6809
The repository shows the 512kb NVRAM implementation.

**Reply by TheEccentricGenius** [reply]
Sorry, been busy here....Fixing a car that was involved in a incident.

I'll look into that schematic hopefully tomorrow, and look into the NVRAM implementation.

**Comment by TheEccentricGenius** [comment]
Sorry for the delay.
Here is the not quite completed main schematic....Haven't done the 64K/512K memory switch yet.
![image](https://github.com/TheEccentricGenius/HD6303RP_EVB/assets/137060094/985a4d38-9d2e-449a-ac95-667de73e344b)
And here is the decode block....I haven't changed the bus yet.
![image](https://github.com/TheEccentricGenius/HD6303RP_EVB/assets/137060094/0258e9f4-6fad-4d85-80ff-b2a34b7616de)

**Comment by TheEccentricGenius** [comment]
I see the resort to SMD for the EEPROM in the SST-6809 board....But there is no other it could really fit on that board.

Overall very nice looking board....When I do the PCB for this I'll try to fit it in the 10cm*10cm space too.

Oh, and I need to add the status LED on my board, and perhaps using the extra latch pins as you suggested.

**Comment by KenWillmott** [comment]
The EEPROM is a PLCC-32 socket, with through plated holes. The chip has side contacts like pins, which makes the EEPROM small and removable. So it assembles like the other components . You're right, it was chosen for the space saving. As an afterthought, it also makes it easy to change the EEPROM instead of reprogramming it. That format of 28C256 is also cheaper, at least from where I'm sitting.

**Comment by KenWillmott** [comment]
I will have a look at your schematic. One last minute thing I added to the SST-6809 - I brought active but unused pins like BSEL0, BSEL1 on your diagram to a pin header, so they can be used for prototyping shortcuts. On mine, another use is for external panel controls if desired. For example, I brought RESET to a pin header, then if the whole system is in a box, you can easily wire a chassis mounted reset switch. Same thing with others like RTS, Z4-Z6. I don't know now what I would need it for, but maybe I will in future.

**Comment by KenWillmott** [comment]
I had a look at the latest schematic file. It looks fine, I didn't have time for a full proof read, but since it's based on tested logic, I focused on the mods to change 6303Y to 6303RP and it is basically okay. If I'm not mistaken, you still run one extended address line to the 6303RP GPIO port, you can free that by moving it to an unused latch pin.

I will follow this with interest, because my plan is to split the CPU to a separate board. I will basically copy the CPU part of your design to such a board. The 6303Y is a great IC, but I'm going to drop it because all the packages I'm getting are too old, the pins are oxidized and I have to manually clean them. I've had my fun with it, and I expect it will be easier to find 6303RP's in good condition.

I am testing the 6809 board by configuring a memory-less 6303Y board and plugging it in as a debug tool. I found out, there is a problem with the EEPROM but I don't know the cause yet. Nothing works when it's installed so it's something serious. NVRAM and paging does pass, but I discovered (the hard way) that many of the DS1210 that I ordered are defective, I guess they are counterfeit so beware. Of course, I also found some other minor layout errors that I can correct with the next run.

I began thinking about physical layout this week and realized I left things too vague. For an SBC implementation, the bus connector location is arbitrary when peripheral cards are installed perpendicular. But to really support board stacking as an alternative, the connector orientation and location on the board must be standardized. That is required to allow peripheral boards to be installed either perpendicular or stacked (or as a "mezzanine" board). For now, I recommend that any 10x10cm board have the bus connector in the same location as the current SST-6809 board.

The game board currently fails to follow those rules, the connector is oriented "backwards", but the next revision of the board will correct that.

**Reply by TheEccentricGenius** [reply]
I'll be offline the next two days....But will try to finish the rest of that schematic next week.

I didn't end up getting 6303Y as it wasn't breadboard friendly and I like testing circuits on breadboards.

Anyway enjoy the week and have a good time.

**Comment by KenWillmott** [comment]
The better alternative to the 6303Y is the 68HC11A1.  After completing work on some other processors, I will build a CPU card for that as well. I managed to install the ASSIST09 monitor in the NVRAM of the SST-6809, which proves the ACIA as well as NVRAM. So I'll be busy applying corrections for another PCB version. At least now, I have a fully working partial system. I'm also experiencing a slowdown due to a busy schedule.

**Reply by TheEccentricGenius** [reply]
Updated the schematics, I haven't done anymore work on the bus though.

BTW, I've found a program that does S19 to binary conversion from STM....Link to [STM32CubeProgrammer](https://www.st.com/en/development-tools/stm32cubeprog.html)
Thanks to a user on the Arduino forum.

**Comment by KenWillmott** [comment]
I looked at your latest PDF, but the latch connections aren't obvious for the reason that KiCad doesn't identify the ins and outs of a bus, you have to net label them all specifically. But assuming you run the latch outputs to the correct RAM address pins, it could be okay. I recommend eliminating the RAM select line from the 6303 GPIO pin and directing that to another latch pin. The GPIO pin can be an input, but the latch is only output and that is all that is required for the job. I have high confidence in everything else because it's already been proven on the 6303Y EVB.

You chose the simplest glue logic which is a good idea for a wired prototype. The two decode circuits that I designed after that have one, and two, extra 74xx IC's respectively. It adds features but also extra pain without any PCB implementation. I suppose you can skip the bus since a proto board won't fit a double row 40 pin connector anyway?

Over here things have gravitated towards a multiple processor platform, not multiprocessing but the ability to choose processors by plugging in a dedicated CPU board. I already have this on my bench, a 6809 board with no processor installed, and a 6809 board converted to a CPU card, plugged into that. I salvaged the boards non-working PLCC EEPROM by piggybacking a DIP EEPROM on top of the RAM and running a few control lines via fly wires. Actually I can also plug a CPU-only 6303Y EVB board in the same way, as an adjunct board. It's the method I used to upload firmware directly to the NVRAM.

Thus I am prepared to make some other CPU board, 6502 for example. I'm mulling over which one would be the most interesting now.

V2.0 of the SST-6809 is in fabrication as of today. It uses the 3rd decode circuit I mentioned, the main difference is that it has 56kb of main memory and only 4kb of EEPROM in the map. The EEPROM is now paged and also jumper selected so that firmware can be swapped for different processors or other reasons. I am not sure how I managed to add yet another IC to a crowded board, but I did. :) After this weekend, I will find some time to upload it so keep an eye out for that.

**Reply by TheEccentricGenius** [reply]
I'll work on cleaning up the PDF then, I was going to ask how things were going over there....But you answered that.

I guess a multiple processor platform is very flexible, what happened to the PLCC EEPROM?
Where you going to include a Z80 design?

**Comment by KenWillmott** [comment]
The new board still has the PLCC EEPROM, hopefully working this time.  I looked at the Z80 data sheet, thinking about how the interface would work. It would definitely require some logic, I don't know if there is any magic pill solution so it's on a back burner for now, as is the CDP1802 which has a similar bus. A 65C02 board could be made with only two parts - a CPU and an oscillator can. So it's tempting to proto that soon.

For 8080 style CPU's it might make more sense to adopt a different bus specification. Right now I'm focused on interoperability so that is a future possibility.

Also if you plug a CPU into a one slot SBC, there is no place for any peripheral boards, so the plan is to make a four slot MAD board (Memory and Decode). One slot for the processor and two or three expansion slots. The M8 bus can support this.

**Reply by TheEccentricGenius** [reply]
OK.
I just mentioned Z80, because they have the eZ80 which is faster.
I guess that MAD board would be good.

How many chips are you planning on supporting?

**Comment by KenWillmott** [comment]
I haven't done a thorough timing analysis, but the system has a maximum clock frequency. The limiting factor now is the ACIA which has a maximum 2MHz speed for a B part. So I won't try to exceed that even if the memory and decode can go faster. In fact, in the big picture, some peripheral IC's can't exceed 1MHz so any system that uses those has to either use a 1Mhz CPU clock, or employ E clock stretching. The game card can run at 2.0MHz so it's not a problem in that regard. The Rockwell 65C02's that I have are 2MHz parts, so perfect for that bus speed.

I definitely plan to support 6809/6309, 6803/6303, and 6502. Then I may deem it worthwhile to bring in 68HC11 and maybe 68008 but I don't want to get too sidetracked with too many processors as the software support will be very time consuming. Also, the game card is hopefully not the last peripheral card - I'd like to make more.

**Reply by TheEccentricGenius** [reply]
Was EmilyJane doing a 68000 board?
Yes, software development would be tedious for a lot of boards.

I hadn't considered the timing speed yet, the eZ80 was just a idea....Now I see it wouldn't really help.
Have a good day.

**Comment by KenWillmott** [comment]
I haven't heard anything from EJ lately. Well, I'll investigate the eZ80  (and the Z80) later. It would be the hardest job by far. The approach taken with the Apple II CP/M card, is to give the Z80 it's own private memory array, separate from the native 6502. Maybe something like that could work.

**Comment by KenWillmott** [comment]
I looked at your schematic again. Your mode pin selection is (P22=1, P21=0, P20=0). That picks mode 4, which is Expanded Multiplexed mode on the 6303. Hitachi made mode 2 and 4 the same, but not so with the Motorola 6803, where mode 4 is Single Chip Test mode. To be uniform and possibly allow the use of Motorola parts, I suggest using mode 2, (P22=0, P21=1, P20=0).

**Reply by TheEccentricGenius** [reply]
I'll update that soon as I can.
Has all been going alright with your development?
I looked a few days ago and didn't notice changes that is why I'm asking.

**Comment by KenWillmott** [comment]
Yes, my documentation is way behind what I'm doing, sorry. I have made a third revision of the 6809 board, V2.1. It's mostly complete but I'm trying to refine the reset circuit. I don't fully trust the simple RC circuit even though it works now. The DS1813 works well but it's a little expensive so I'm hunting other parts and trying them out. There were minor tweaks, like supporting both 128k and 512k RAM in the same socket with no jumper changes. Edit - all you have to do is run A17 from the latch through an inverter before going to pin 30 on the NVRAM.  On the very, very back end I found square wood pieces at the dollar store, which can make a kind of sandwich stack housing for the boards using standoffs. Those were sanded, drilled and given a Urethane finish.

I did a lot of work to standardize the board footprint, connector and standoff spacing, so they can stack properly. Also I have two new boards, a 6522 VIA with a prototyping area, and a small plug in 6502 adapter. Those are in fab, I should have them next week to try. I prioritized the VIA because I realized the 6809 board has no timer, which hobbles most real time applications.

In the software department, I wrote some 6809 code to write, lock, and unlock the EEPROM in place.

Originally I imagined the 6303 as another plug in processor similar to the 6502 board I made, but now I'm more leaning towards making another main board using it in place of the 6809.

**Reply by TheEccentricGenius** [reply]
I see, a lot of work going on down there.
An off topic question, do you do a lot of woodwork or is that just another hobby?

I don't know if I have a spare inverter on the board, there is none IIRC....Although there is an obvious advantage with a inverter.

The EEPROM unlock/lock/write sounds like a handy function, unfortunately progress is slow down here due to a shed being constructed.

**Comment by KenWillmott** [comment]
Not a lot, but I frequently make wooden cases for my projects - especially the digital clocks. It was a hobby when I was making hardwood speaker cabinets. But I don't do that any more, I have no tools or space for it now. You're right - your decode circuit has no left over gates. You can just connect that pin to Vcc instead, for the 128k parts.

I'm struggling with the task of avoiding counterfeit parts. Many of the parts I need are not stocked by distributors here, or probably anywhere. So I'm depending on the NOS and pull market. Those are fine with me, but I can't tolerate re-branded parts and especially when it's an NMOS part sold as a CMOS part. I've been reading some sad tales of MC6809's from Asia - horrendously careless refurbishing and relabelling of parts. That's why I'm leaning strongly into Hitachi parts - my theory, which seems to be working for me so far, is that there are too few sold to make it worth the counterfeiter's time.

**Comment by KenWillmott** [comment]
I've updated the SST6809 project. I wonder if you can try and load it in KiCad and see if I missed anything? I diverted my attention on the 6303 to the Tandy TRS-80 MC-10 which uses the MC6803 processor. I found out, it won't work unless all the mode pins are correct, even though some for mode 3 don't seem to apply directly. This contradicts my earlier guess that you would only need one boot mode pin.

I found, there is a lot you can learn about the 6803/6303 from the MC-10. Even though, the designers threw away some of the nicer features in order to make it cheaper and more suited to a 1980's home computer. I'm trying to fit an HD63C03Y into a PI housing as my 6303 project, but I've gotten sidetracked with an MC-10 memory expansion and other projects.

**Reply by TheEccentricGenius** [reply]
Yes, sure thing.
Sorry about the lack of activity here, Chess and my job take a lot of my time now days.
Life seems to have a way of pausing hobbies.

**Comment by KenWillmott** [comment]
It's okay, some of my projects took 10 years. Same kind of reasons. Thanks for kicking ideas around with me.

**Reply by TheEccentricGenius** [reply]
I will still be here to kick around ideas....

**Reply by TheEccentricGenius** [reply]
All appears to be correct from my look at the project, however you may want to include the bus and decode schematics in the project too.
I got some errors because those two files aren't there.

I have been studying some old computers in my spare time some of the time too, it is quite interesting to read the background stories behind them too.

Currently I have the HD6303RP and all the necessary circuitry on two breadboards, with the EEPROM burned with the monitor you supplied.
However I am up to wiring everything up, and because I want it to be semi-permeant I have to cut wire to suit.
Unfortunately cutting the wire is a bit tedious so I haven't done it yet.

I actually have been designing a whole computer from logic gates, so that distraction hasn't helped either.
